corehq/apps/cloudcare/views.py

```diff
@@ -350,7 +351,14 @@ def filter_cases(request, domain, app_id, module_id, parent_id=None):
     cases = filter(lambda c: c.type == case_type, cases)
     cases = [c.get_json(lite=True) for c in cases if c]
 
-    return json_response(cases)
+    response = {'cases': cases}
+    if requires_parent_cases:
+        parent_ids = map(lambda c: c['indices']['parent']['case_id'], cases)
+        parents = [CommCareCase.wrap(doc) for doc in iter_docs(CommCareCase.get_db(), parent_ids)]
```

Comment:
not sure if this would be considered a premature optimization, but I think we can check our local `cases` before going back to the DB to get them in the case where we were doing a full footprint already (or our parent type is the same as the child type)